src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/model/PrisonRepository.kt

```diff
@@ -18,7 +18,7 @@ interface PrisonRepository : JpaRepository<Prison, String>, JpaSpecificationExec
   fun findAllByPrisonIdIsIn(ids: List<String>): List<Prison>
 
   @Query(
-    "SELECT new uk.gov.justice.digital.hmpps.prisonregister.resource.dto.PrisonNameDto(p.prisonId, p.name) FROM Prison p WHERE (:active IS NULL OR p.active = :active) ORDER BY p.name",
+    "SELECT new uk.gov.justice.digital.hmpps.prisonregister.resource.dto.PrisonNameDto(p.prisonId, p.name) FROM Prison p WHERE (:active IS NULL OR p.active = :active) AND (:name IS NULL OR p.name = :name)ORDER BY p.name",
```

Comment:
I don't like this approach, we would normally use a Specification for filters like this.  There isn't really a need to write JQL - you can simply load the entity and then map to a Dto.  The JPA entity should have lazy loading on the collections so no new joins will occur.

Reply:
I will check about this